doc/DHCPv6_relay/DHCPv6-relay-agent-High-Level-Design.md

```diff
@@ -0,0 +1,233 @@
+  - Hop-count in Relay-forward message is set to 0.
+  - If Relay Agent were to relay a message from a relay agent, it checks if the hop-count in the message is greater than or equal to HOP\_COUNT\_LIMIT, and discard if so. Else, hop\_count is incremented by 1.
+3. DHCPv6 server received the SOLICIT message, refers to the Relay Agent IP and select an IP address to allocate to the DHCPv6 client.
+4. The DHCPv6 server constructs a RELAY-REPLY message that embeds the ADVERTISE messages, and sends it to the DHCPv6 relay agent.
+5. DHCPv6 relay agent extracts ADVERTISE message from RELAY-REPLY message and forwards it to the client.
+6. DHCPv6 client receives ADVERTISE message and relays a REQUEST message to the DHCPv6 relay agent.
+7. DHCPv6 relay agent constructs REQUEST message into a RELAY-FORWARD message, and relays to DHCPv6 server.
+8. DHCPv6 server receives the REQUEST message, and sends a REPLY message to the relay agent. Server creates a Relay-reply message that includes a Relay Message option containing the the REPLY message and sends it to the relay agent.
+9. DHCPv6 relay agent extracts message and relays the message to the address contained in the peer-address field of the Relay-reply message.
+10. DHCPv6 client receives the REPLY message that contains the desired IP address.
+
+![image](https://user-images.githubusercontent.com/42761586/117859842-5fd13b00-b244-11eb-9297-c2674d128dd9.png)
+
+# Requirements
+
+- Configured and running DHCPv6 client and server
+- Connectivity between the relay agent and DHCPv6 server
+- Configure one or more IP helper addresses for specified VLANs to forward DHCPv6 requests to DHCPv6 servers on other subnets.
+- Client UDP port:546
+- Server and Relay Agent UDP port: 547
+
+# Topology
+
+![image](https://user-images.githubusercontent.com/42761586/117859824-58aa2d00-b244-11eb-991f-7ad8759f6612.png)
+
+# Design
+
+# CLI and Usage
+
+-show dhcp6relay_counters
+
+-sonic-clear dhcprelay_counters
+
+-enable/Disable option 79
+
+-enable/Disable use-loopback-address (for dual tor)
+
+-show/config ip helpers
+
+# DHCPRELAY counter
+
+Keeps count of all relay Messages:
+SOLICIT
+ADVERTISE
+REQUEST
+CONFIRM
+RENEW
+REBIND
+REPLY
+RELEASE
+DECLINE
+RELAY-FORWARD
+RELAY-REPLY
+
+# CONFIG DB schema
+
+<pre>
+DHCP|intf-i|dhcpv6\_servers: [&quot;dhcp-server-0&quot;, &quot;dhcp-server-1&quot;, ...., &quot;dhcp-server-n-1&quot;]
+
+DHCP|intf-i|dhcpv6\_options: [&quot;79&quot;]
+</pre>
+
+# YANG Model schema
+
+sonic-dhcpv6-relay.yang
+<pre>
+module DHCP  
+    container DHCP {  	
+        list VLAN_LIST {
+    		key name;
+   		    leaf name {
+    			type string;
+  		    }
+   		    leaf dhcpv6_servers {
+     		    type inet6:ip-address;
+  		    }
+		    leaf options {
+			    type uint16;
```

Comment:
does this only allow one option, what if we have other options? I suggest to define option79 true/false. some options needs parameter. it is difficult to define a universal way to define options.

Reply:
Updated option 79 to have a bool type! Future options added can define its own parameter.